### PR TITLE
card: process that abstracts car interface and CAN

### PR DESCRIPTION
**Pull request by sshane**

### CODE REVIEW

**Comment by sshane** [comment]
whoops

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -439,7 +440,7 @@ def data_sample(self):
 
     # All pandas not in silent mode must have controlsAllowed when openpilot is enabled
     if self.enabled and any(not ps.controlsAllowed for ps in self.sm['pandaStates']
-           if ps.safetyModel not in IGNORED_SAFETY_MODES):
+                            if ps.safetyModel not in IGNORED_SAFETY_MODES):
```
Comment (sshane) [inline selfdrive/controls/controlsd.py:443]:
```suggestion
           if ps.safetyModel not in IGNORED_SAFETY_MODES):
```

**Comment by sshane** [comment]
NEW controlsd & card control flow:

- controlsd->data_sample: waiting for carState
# - controlsd->initialized = True
# - card-> initialize
- card->state_update: receive can and create CS, gets latest CC
- card->state_publish: publish CS, CP, CO
  - I think this is incorrect as at this point we haven't updated CO or actuated anything so controlsd will get the doubly previous CO
- card->controls_update: actuate CC and update CO

**Comment by sshane** [comment]
NEWER controlsd & card control flow:

- controlsd->data_sample: waiting for carState
- card->state_update: receive can and create CS, gets latest CC
  - I think this is incorrect as at this point we haven't updated CO or actuated anything so controlsd will get the doubly previous CO
- card->controls_update: actuate CC and update CO
- card->state_publish: publish CP, CO, CS
- controlsd: gets latest CS and CC

**Comment by sshane** [comment]
There is some controlsState diff, that is expected:

1) old routes, such as FORD, had CS.events encompass all openpilot alerts rather than just be car events. This leads to duplicate events in onroadEvents until the route is replaced
2) carControl will be delayed by one frame

File: selfdrive/controls/controlsd.py [hunk 2]
```diff
@@ -82,6 +84,7 @@ def __init__(self, CI=None):
     self.camera_packets = ["roadCameraState", "driverCameraState", "wideRoadCameraState"]
 
     self.log_sock = messaging.sub_sock('androidLog')
+    self.car_state_sock = messaging.sub_sock('carState', timeout=20)
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:87]: let's add a TODO here, don't think we want to keep it this way

File: selfdrive/car/card.py [hunk 3]
```diff
@@ -18,23 +18,26 @@
 
 REPLAY = "REPLAY" in os.environ
 
+State = log.ControlsState.OpenpilotState
 EventName = car.CarEvent.EventName
 
 
-class CarD:
+class Car:
```
Comment (adeebshihadeh) [inline selfdrive/car/card.py:25]: I think `card` is small enough that wrapping in this class makes it less readable with all the `self` and misdirection

Comment (sshane) [reply selfdrive/car/card.py:25]: Will do after this is merged

**Pull request merged**